From 28cf0b010336af113b1397e42428a7c17903a4c1 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Tue, 05 Nov 2024 15:44:44 +0200
Subject: [PATCH] [FSW-8694] fs1: remove unnecessary memcopy

This patch removes unnecessary memcopy which
is used in fake fuse initialization, this can
be done by directly assigning address to
required data pointer.

Change-Id: Ia4c2a15267aa759ea7dce8bd0c752bfc70719b6e
---

diff --git a/common/boot_desc.c b/common/boot_desc.c
index e4cbb6f..c900a30 100644
--- a/common/boot_desc.c
+++ b/common/boot_desc.c
@@ -352,7 +352,7 @@
 	uint32_t fake_fuse_base, fake_fuse_boot_ptr_size;
 	struct hl_fw_img_desc *fake_fuse_img_desc;
 	struct hl_fw_manifest_constants spi_manifest_consts;
-	struct hl_fw_boot_ptrs fake_fuse_boot_ptr;
+	struct hl_fw_boot_ptrs *fake_fuse_boot_ptr;
 	uint32_t *offset_array_base = NULL;
 
 	spi_manifest_consts.dest_addr_dma = SPI_MANIFEST_IMG_2_DMA_DEST;
@@ -379,24 +379,23 @@
 		return BOOT_ROM_FAIL;
 	}
 
-	memcpy(&fake_fuse_boot_ptr, (uint32_t *)(spi_manifest_consts.dest_addr_arc),
-		sizeof(struct hl_fw_boot_ptrs));
+	fake_fuse_boot_ptr = (struct hl_fw_boot_ptrs *)(spi_manifest_consts.dest_addr_arc);
 
-	if (fake_fuse_boot_ptr.magic.val != (FAKE_FUSE_BOOT_PTR_MAGIC_NAME | FW_SPI_BOOT_PTR_SUB)) {
+	if (fake_fuse_boot_ptr->magic.val != (FAKE_FUSE_BOOT_PTR_MAGIC_NAME | FW_SPI_BOOT_PTR_SUB)) {
 		boot_rom_status_update(PROGRESS_INIT_EFUSE_SPI_MAGIC_FAIL);
 		return BOOT_ROM_FAIL;
 	}
 
 	offset_array_base = (uint32_t *)(spi_manifest_consts.dest_addr_arc +
-		fake_fuse_boot_ptr.offset2array);
+		fake_fuse_boot_ptr->offset2array);
 
-	if ((offset_array_base == NULL) || (fake_fuse_boot_ptr.count == 0)) {
+	if ((offset_array_base == NULL) || (fake_fuse_boot_ptr->count == 0)) {
 		boot_rom_status_update(PROGRESS_INIT_EFUSE_SPI_DESC_COUNT_FAIL);
 		return BOOT_ROM_FAIL;
 	}
 
 	if (die_id_index == MAX_DIE_COUNT) {
-		for(index = 0; index < fake_fuse_boot_ptr.count; index++) {
+		for(index = 0; index < fake_fuse_boot_ptr->count; index++) {
 			fake_fuse_img_desc = (struct hl_fw_img_desc *)(spi_manifest_consts.dest_addr_arc +
 				offset_array_base[index]);
 
